src/backends/plonky2/main.rs

```diff
@@ -0,0 +1,278 @@
+use crate::backends::plonky2::basetypes::{Hash, Value, D, EMPTY_HASH, EMPTY_VALUE, F, VALUE_SIZE};
+use crate::backends::plonky2::common::{
+    CircuitBuilderPod, OperationTarget, StatementTarget, ValueTarget,
+};
+use crate::backends::plonky2::mock_signed::MockSignedPod;
+use crate::backends::plonky2::primitives::merkletree::MerkleProofExistenceCircuit;
+use crate::backends::plonky2::primitives::merkletree::{MerkleProof, MerkleTree};
+use crate::middleware::{
+    hash_str, AnchoredKey, Operation, Params, PodType, Statement, ToFields, KEY_TYPE, SELF,
+    STATEMENT_ARG_F_LEN,
+};
+use anyhow::Result;
+use itertools::Itertools;
+use plonky2::field::extension::Extendable;
+use plonky2::hash::hash_types::RichField;
+use plonky2::{
+    field::types::Field,
+    hash::{
+        hash_types::{HashOut, HashOutTarget},
+        poseidon::PoseidonHash,
+    },
+    iop::{
+        target::{BoolTarget, Target},
+        witness::{PartialWitness, WitnessWrite},
+    },
+    plonk::circuit_builder::CircuitBuilder,
+};
+use std::collections::HashMap;
+
+const MD: usize = 32;
```

Comment:
does this stand for "max depth" of the merkletree?

Reply:
yes! I'll add a comment